contenttype/middleware.go

```diff
@@ -22,8 +22,19 @@ func NewMiddleware() Middleware {
 	return &middleware{}
 }
 
+type contentTypeResponseWriter struct {
+	http.ResponseWriter
+	r *http.Request
+}
+
+func (erw *contentTypeResponseWriter) Write(b []byte) (int, error) {
+	if b != nil && erw.ResponseWriter.Header().Get(contentTypeHeaderKey) == "" {
+		erw.ResponseWriter.Header().Set(contentTypeHeaderKey, defaultContentType)
```

Comment:
Headers shouldn't be set in the `Write` method, because it should expect header to already be sent.